Source/CombatExtended/CombatExtended/Comps/CompFragments.cs

```diff
@@ -84,17 +83,30 @@ public void Throw(Vector3 pos, Map map, Thing instigator, float scaleFactor = 1)
                     Log.Warning("CombatExtended :: Tried to throw fragments out of bounds");
                     return;
                 }
-                var projCE = parent as ProjectileCE;
-                var edifice = pos.ToIntVec3().GetEdifice(map);
-                var edificeHeight = edifice == null ? 0f : new CollisionVertical(edifice).Max;
-                var height = projCE != null ? Mathf.Max(edificeHeight, projCE.Height) : edificeHeight;
+		float height;
+		FloatRange fragXZAngleRange;
+		if (parent is ProjectileCE projCE)
+		{
+		    height = projCE.Height;
+		    fragXZAngleRange = new FloatRange(projCE.shotRotation + PropsCE.fragXZAngleRange.min, projCE.shotRotation + PropsCE.fragXZAngleRange.max);
+		}
+		else
+		{
+		    height = 0;
+		    fragXZAngleRange = PropsCE.fragXZAngleRange;
+		}
+                if (pos.ToIntVec3().GetEdifice(map) is Building edifice)
+		{
+		    var edificeHeight = new CollisionVertical(edifice).Max;
+		    height = Mathf.Max(height, edificeHeight);
```

Comment:
I'm unsure whether or not this change would work, but perhaps give a `0.001f` offset to the height? It may help with all fragments being stopped by embrasures on explosion.

Reply:
I was thinking more about just raising it by `0.001f` overall so that explosive projectiles which do land on cover spawn fragments which will not be intercepted by the same cover.